Factoring out the quantized mappings

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23351 Factoring out the quantized mappings**

Differential Revision: [D16470907](https://our.internmc.facebook.com/intern/diff/D16470907)

Commit:
Factoring out the quantized mappings

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -4,6 +4,9 @@
 import torch.nn.qat as qat
```
then we can remove import of nnq and qat?

jerryzh168 commented on torch/quantization/_mappings.py:
```diff
@@ -0,0 +1,27 @@
+
+import torch
+
+from .quantize import QuantStub
```
do we have circular dependency here?

jerryzh168 commented on torch/quantization/_mappings.py:
```diff
@@ -0,0 +1,27 @@
+
+# Map for swapping float module to quantized ones.
+_DEFAULT_MODULE_MAPPING = {
+    torch.nn.Linear: torch.nn.quantized.Linear,
```
we can import nn, nnq and qat here to simply namespaces.

Commit:
Update on "Factoring out the quantized mappings"


Differential Revision: [D16470907](https://our.internmc.facebook.com/intern/diff/D16470907)

Commit:
Update on "Factoring out the quantized mappings"


Differential Revision: [D16470907](https://our.internmc.facebook.com/intern/diff/D16470907)

zafartahirov commented:
Closing in favor of #27164
